src/Json.Schema.ToDotNet.UnitTests/CommandLineTests.cs

```diff
@@ -0,0 +1,55 @@
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=Int", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=int", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=INT", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=iNT", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=long", (GenerateIntegerOption?)GenerateIntegerOption.Long, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=biginteger", (GenerateIntegerOption?)GenerateIntegerOption.BigInteger, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=auto", (GenerateIntegerOption?)GenerateIntegerOption.Auto, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=unknown", (GenerateIntegerOption?)null, "generate-integer-as"),
+            };
+
+            foreach ((string argsString, GenerateIntegerOption? expectedGenerateIntegerAs, string expectedErrorParameter) testCase in testCases)
+            {
+                var args = testCase.argsString.Split(' ');
+                var parser = new Parser(cfg => cfg.CaseInsensitiveEnumValues = true).ParseArguments<Options>(args)
+                    .MapResult(
+                    options =>
+                    {
+                        testCase.expectedGenerateIntegerAs.Should().NotBeNull();
```

Comment:
The first test case to failure its conditions fails the test as a whole, and the developer will not know if any other testcases in the test are broken too.  What does the test failure output look like? Is it clear what test case failed?

Recommend making it do its verification checks and append to a failure string builder, and at the end of the test emits all test case failures in a clear manner that is easy to debug and ensure we tested everything.

Reply:
Yes! Fixed.